src/libs/ReportActionsUtils.ts

```diff
@@ -261,7 +261,7 @@ function isConsecutiveActionMadeByPreviousActor(reportActions: ReportAction[] |
 /**
  * Checks if a reportAction is deprecated.
  */
-function isReportActionDeprecated(reportAction: OnyxEntry<ReportAction>, key: string): boolean {
+function isReportActionDeprecated(reportAction: OnyxEntry<ReportAction>, key: string | number): boolean {
```

Comment:
why are these string or number now? I'd thought the API should be returning reportActionIDs as strings all the time so if they aren't we need to fix that in the API.

Reply:
Here `actionKey` is `number` that's why. Which is passed to `shouldReportActionBeVisible`  which invoked `isReportActionDeprecated` with the same type.

https://github.com/Expensify/App/blob/1c31bf72ec7d77d01fdb6731d9fb6e94b8689af3/src/libs/SidebarUtils.js#L32-L36

Reply:
Hmm think I understand. 

so `actionKey` is an int because in that code for `_.filter` we're looping over an array which is going to be keyed by an integer?

Reply:
Yes `index` is `number`!

Reply:
hm okay. 

It looks like this is mostly there just for checking if the deprecated sequenceNumbers exist which I think should be fully killed by now but I don't think we should do too much in this refactor so I think it's probably fine to leave as is for now.

but after we merge this we should probably clean up that code, i don't think we need to pass `key` ever and should be able to only pass `reportAction` but I'm no the most involved on this side of things either.